(Feature) New style for xDai network

gabitoesmiapodo opened the pull request:
Closes #97 

**Description:**

Added support for xDai Network's styles.

Updated styles for the rest of the networks.

Added some standard components taken from other POA Network's dApps, in hopes of having a more coherent and consistent ecosystem.

**Some screenshots:**

<img width="1040" alt="screen shot 2019-01-16 at 17 22 14" src="https://user-images.githubusercontent.com/4015436/51277291-f1527a00-19b5-11e9-9405-e3c2d6672f42.png">
<img width="1035" alt="screen shot 2019-01-16 at 17 22 24" src="https://user-images.githubusercontent.com/4015436/51277293-f1527a00-19b5-11e9-87e0-536fd4a4f283.png">
<img width="335" alt="screen shot 2019-01-16 at 17 26 33" src="https://user-images.githubusercontent.com/4015436/51277294-f1527a00-19b5-11e9-8649-5898f05a8b66.png">
<img width="338" alt="screen shot 2019-01-16 at 17 26 44" src="https://user-images.githubusercontent.com/4015436/51277295-f1527a00-19b5-11e9-92cd-fb5339e45e04.png">
<img width="1037" alt="screen shot 2019-01-16 at 17 27 09" src="https://user-images.githubusercontent.com/4015436/51277296-f1eb1080-19b5-11e9-826d-d21e75cf70ab.png">
<img width="1041" alt="screen shot 2019-01-16 at 17 33 02" src="https://user-images.githubusercontent.com/4015436/51277297-f1eb1080-19b5-11e9-95d8-c43968b11d16.png">
<img width="1039" alt="screen shot 2019-01-16 at 17 33 11" src="https://user-images.githubusercontent.com/4015436/51277298-f1eb1080-19b5-11e9-8401-5a97fc739d67.png">

gabitoesmiapodo committed:
(refactor) file and folders' structure / scaffolding - (fix) eslint + prettier - (add) npm start watches scss now

gabitoesmiapodo committed:
(remove) unused files - (add) base components: header, footer, menu, etc.

gabitoesmiapodo committed:
(remove) unused files - (fix) validators loading - (fix) dApp name in logo - (change) link for navlink

gabitoesmiapodo committed:
(add) validator component styles

gabitoesmiapodo committed:
(fix) logos - (add) validator component - (add) physical address component

gabitoesmiapodo committed:
(add) finalize / confirm buttons components.

gabitoesmiapodo committed:
(add) Validator title icons.

gabitoesmiapodo committed:
(remove) unused files - (add) confirmed address icon's network colors

gabitoesmiapodo committed:
(remove) unused files

gabitoesmiapodo committed:
(fix) Section title's display.

gabitoesmiapodo committed:
(add) styles for set metadata form

gabitoesmiapodo committed:
(chore) removed temporary code - (fix) sweet alerts z-index

vbaranov commented on src/App.js:
```diff
@@ -258,102 +259,101 @@ class App extends Component {
-    let loader = this.state.loading ? <Loading /> : ''
 
-    const isDaiNetwork = netId === helpers.netIdByName('dai')
+    let loader = this.state.loading ? <Loading /> : ''
```
Does `loader` object change after assignment? if no, let's change to `const`

vbaranov commented on src/components/Validator/index.js:
```diff
@@ -0,0 +1,96 @@
+      updatedDate
+    } = this.props
+
+    if (netId === helpers.netIdByName('dai') && !createdDate) {
```
`dai` and other networks names are meet in different places in the code. Let's define constants for them.

vbaranov reviewed (changes requested):
- There is no visible loader when switching to *Pending changes* or *All* back
- Impossible to search by full name (to be truth, it is not the issue of this PR, but it would be nice if it could be fixed) <img width="1146" alt="screenshot 2019-01-17 at 14 28 00" src="https://user-images.githubusercontent.com/4341812/51315827-27e6ce00-1a64-11e9-8c19-15ded2415346.png">
- change warning text **The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask** to **The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask/Nifty Wallet**

vbaranov commented:
@gabitoesmiapodo 

Also,

- if to open directly `/set` page, the loader is shown permanently <img width="824" alt="screenshot 2019-01-17 at 15 07 34" src="https://user-images.githubusercontent.com/4341812/51317821-a5610d00-1a69-11e9-8bf1-9843c57b1b0d.png">
- wrong loader color for `Sokol` network by clicking on *Update metadata* button <img width="827" alt="screenshot 2019-01-17 at 15 09 00" src="https://user-images.githubusercontent.com/4341812/51318031-346e2500-1a6a-11e9-97d2-bd727fe3773b.png">

gabitoesmiapodo committed:
(fix) change warning text / change loader to const  / define constants for network's names /

gabitoesmiapodo committed:
(fix) loading component issues

gabitoesmiapodo committed:
(fix) make possible to search by full name

gabitoesmiapodo commented:
@vbaranov 

1) There is no visible loader when switching to Pending changes or All back -> Fixed.

2) Impossible to search by full name (to be truth, it is not the issue of this PR, but it would be nice if it could be fixed) -> I think the way this is implemented it can get quite complicated easily (I mean, it could be improved), but I believe it's working as expected now. Please check it out.

3) change warning text The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask to The key is not valid voting Key or you are connected to the wrong chain! Please make sure you have loaded correct voting key in Metamask/Nifty Wallet -> Changed it.

4) Does loader object change after assignment? if no, let's change to const -> Fixed.

5) dai and other networks names are meet in different places in the code. Let's define constants for them. -> Done.

6) if to open directly /set page, the loader is shown permanently -> Fixed.

7) wrong loader color for Sokol network by clicking on Update metadata button -> Fixed.

varasev commented:
I will also review that.

vbaranov commented:
@gabitoesmiapodo 

![screenshot 2019-01-18 at 19 57 59](https://user-images.githubusercontent.com/4341812/51401290-6f03ba80-1b5b-11e9-8df0-b957522d829b.png)

gabitoesmiapodo commented:
@vbaranov Sorry, I was on vacation until yesterday.

Yes, I'm working on it.

Any other feedback from you or @varasev ?

varasev commented:
@gabitoesmiapodo I didn't review this yet, but I'll do this after the error above is fixed.

vbaranov commented:
>Any other feedback from you

Oh, thanks. No, as far as I can see this is the last issue blocking the merging from my perspective.

gabitoesmiapodo committed:
(fix) wrong amount of fields sent to 'set metadata' form

gabitoesmiapodo commented:
@vbaranov 

It should be fixed now.

I think it had something to do with me not rendering unnecessary components when they were not needed (instead of just hiding them), so I changed that and now they are hidden / shown depending on the situation (as they were before I changed that).

vbaranov commented:
@gabitoesmiapodo I still experiencing this issue in https://github.com/poanetwork/poa-dapps-validators/pull/103/commits/a8db3ac471d13900a471998ca53c323e15eaf952

gabitoesmiapodo commented:
@vbaranov 

OK, any leads about what could be causing it? I'm not an actual validator (and don't have a validator key) and can't test the problem in real conditions.

I'm not sure about what could be causing it, all required nine fields are there.

varasev commented:
`ValidatorMetadata.createMetadata` method was changed for `xDai` network and now [it expects 9](https://github.com/poanetwork/poa-chain-spec/blob/dai/abis/ValidatorMetadata.abi.json#L398) parameters instead of 7. However, we have previous `ValidatorMetadata.createMetadata` deployed in Sokol and Core, so for these networks it expects [7 parameters](https://github.com/poanetwork/poa-chain-spec/blob/sokol/abis/ValidatorMetadata.abi.json#L60).

varasev commented:
The corresponding changes were introduced into `Validators DApp` in https://github.com/poanetwork/poa-dapps-validators/pull/98 and https://github.com/poanetwork/poa-dapps-validators/pull/102.

gabitoesmiapodo committed:
Merge branch 'core' into feature/#87-new-style-for-xdai-network

* core:
  Update README.md

gabitoesmiapodo committed:
(fix) revert last changes to app.js

gabitoesmiapodo commented:
@vbaranov 

Okay, reverted the changes for this section.

I hope it's fine now.

vbaranov reviewed (approved):
LGTM

varasev commented:
Please don't merge this yet, I'll also check.

gabitoesmiapodo commented:
@varasev Sure, no problem.
